File: ServiceBooking.API/Services/ClientKey.cs
namespace ServiceBooking.API.Services;

/// <summary>
/// Parses the <c>{clientKey}</c> route segment API_CONTRACT_CYCLE5.md §44/§45 introduces — a registered
/// client's <c>userId</c>, or <c>phone:79991234567</c> (canonical) for a guest. One place this format is
/// understood, so every new salon-facing endpoint that takes a client key (photo consent, health note,
/// health consent — and whatever the next one turns out to be) parses it identically.
/// </summary>
public static class ClientKey
{
    private const string PhonePrefix = "phone:";

    public static bool IsPhone(string clientKey) => clientKey.StartsWith(PhonePrefix, StringComparison.Ordinal);

    public static string ExtractPhone(string clientKey) => clientKey[PhonePrefix.Length..];
}

File: frontend/src/utils/clientKey.ts
/**
 * API_CONTRACT_CYCLE5.md §44.1, §45.1 — `clientKey` is a `userId` for a registered client, or the
 * canonical `phone:79991234567` form for a guest. Shared so every call site (health note, photo
 * consent) builds the same key from the same client shape.
 */
export function getClientKey(client: { clientId: string | null; guestPhone: string | null }): string {
  if (client.clientId) return client.clientId
  return `phone:${client.guestPhone ?? ''}`
}
